cmd/utils.go

```diff
@@ -15,378 +15,411 @@
+	HealthInfoSignature SigDetails `json:"healthInfoSignature"`
+	VexInfoSignature    SigDetails `json:"vexInfoSignature"`
+}
+
+type SigDetails struct {
+	CertInfo struct {
+		Cert      string `json:"cert"`
+		CertChain string `json:"certChain"`
+	} `json:"certInfo"`
+	Digest []struct {
+		Digest    string `json:"digest"`
+		Algorithm string `json:"Algorithm"`
+	} `json:"digest"`
+	Signature []struct {
+		Signature string `json:"signature"`
+		Algorithm string `json:"Algorithm"`
+	} `json:"signature"`
+}
 
 // The 'sbom' key in the buildInfo.json contains package
 // info from where the signature zip URL is extracted.
 type sbom struct {
-    Packages []struct {
-        Spdxid                string `json:"SPDXID"`
-        ExternalRefs     	  []struct {
-            ReferenceCategory	string `json:"referenceCategory"`
-            ReferenceLocator  	string `json:"referenceLocator"`
-        } `json:"externalRefs"`
-    } `json:"packages"`
+	Packages []struct {
+		Spdxid       string `json:"SPDXID"`
+		ExternalRefs []struct {
+			ReferenceCategory string `json:"referenceCategory"`
+			ReferenceLocator  string `json:"referenceLocator"`
+		} `json:"externalRefs"`
+		Annotations []struct {
+			Comment string `json:"comment"`
+		} `json:"annotations"`
+	} `json:"packages"`
 }
 
-
 func downloadFromGCS(ctx context.Context, serviceAccountKeyFilePath string, bucketName string, objectName string, filePath string) error {
-    // Authenticate using the service account key file.
-    client, err := storage.NewClient(ctx, option.WithCredentialsFile(serviceAccountKeyFilePath))
-    if err != nil {
-        return fmt.Errorf("Failed to authenticate to GCS: %v", err)
-    }
-
-    reader, err := client.Bucket(bucketName).Object(objectName).NewReader(ctx)
-    if err != nil {
-        return fmt.Errorf("Failed to open object: %v", err)
-    }
-    defer reader.Close()
-
-    file, err := os.Create(filePath)
-    if err != nil {
-        return fmt.Errorf("Failed to create file: %v", err)
-    }
-    defer file.Close()
-
-    if _, err := io.Copy(file, reader); err != nil {
-        return fmt.Errorf("Failed to download the file: %v", err)
-    }
-
-    // Close the client.
-    client.Close()
-
-    return nil
+	// Authenticate using the service account key file.
+	client, err := storage.NewClient(ctx, option.WithCredentialsFile(serviceAccountKeyFilePath))
+	if err != nil {
+		return fmt.Errorf("failed to authenticate to GCS: %v", err)
+	}
+
+	reader, err := client.Bucket(bucketName).Object(objectName).NewReader(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to open object: %v", err)
+	}
+	defer reader.Close()
+
+	file, err := os.Create(filePath)
+	if err != nil {
+		return fmt.Errorf("failed to create file: %v", err)
+	}
+	defer file.Close()
+
+	if _, err := io.Copy(file, reader); err != nil {
+		return fmt.Errorf("failed to download the file: %v", err)
+	}
+
+	// Close the client.
+	client.Close()
```

Comment:
can we move this client.Close to line 115 and also add defer?

Reply:
done